### PR TITLE
cmd/atlas: initial support for runtimevar as data source

### PR DESCRIPTION
**Pull request by a8m**
More examples and docs will be added in the next PRs.

```hcl
data "runtimevar" "pass" {
  url = "gcpsecretmanager://projects/<project-name>/secrets/<secret-name>"
}

env "local" {
  url = "mysql://root:${data.runtimevar.pass}@host:port/database"
}
```

### CODE REVIEW

**Review by rotemtam — APPROVED** [review]
Very excited to use this !!

File: cmd/atlas/internal/cmdext/cmdext.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+// DataSources exposes the data sources provided by this package.
+var DataSources = []schemahcl.Option{
+	schemahcl.WithDataSource("sql", QuerySrc),
+	schemahcl.WithDataSource("runtimevar", RuntimeVarSrc),
+}
+
+// RuntimeVarSrc exposes the gocloud.dev/runtimevar as a schemahcl datasource.
+//
+//	data "runtimevar" "pass" {
+//	  url = driver://path?query=param
```
Comment (giautm) [inline cmd/atlas/internal/cmdext/cmdext.go:37]:
```suggestion
//	  url = "driver://path?query=param"
```

File: cmd/atlas/internal/cmdext/cmdext.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/gohcl"
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/zclconf/go-cty/cty"
+	"gocloud.dev/runtimevar"
+	_ "gocloud.dev/runtimevar/awssecretsmanager"
+	_ "gocloud.dev/runtimevar/constantvar"
+	_ "gocloud.dev/runtimevar/gcpsecretmanager"
+	_ "gocloud.dev/runtimevar/httpvar"
+)
```
Comment (giautm) [inline cmd/atlas/internal/cmdext/cmdext.go:26]: Include all secrets? hashicorp/vault, azure?

Comment (a8m) [reply cmd/atlas/internal/cmdext/cmdext.go:26]: added filevar. blob depends on all other blob drivers, which I prefer to avoid atm.

**Pull request merged**